### PR TITLE
[DPE-5248] Add pgAudit

### PR DESCRIPTION
**Pull request by marceloneppel**
## Issue
The pgAudit plugin/extension is available in the latest revision of the Charmed PostgreSQL snap, but not in this charm.

## Solution
Expose pgAudit as another plugin/extension charm config option and enable it when requested by:
* Using the new snap, which contains the pgAudit plugin/extension.
* Adding `pgaudit` to the PostgreSQL `shared_preload_libraries` parameter.
* Configuring the requested parameters from [DPE-2573](https://warthogs.atlassian.net/browse/DPE-2573) on `lib/charms/postgresql_k8s/v0/postgresql.py`.

Added new unit and integration tests.

Also, some adjustments were made in the plugin retrieval, which is used in the `config-changed` and `database-requested` hook handlers.

This is a port of https://github.com/canonical/postgresql-k8s-operator/pull/688.

[DPE-2573]: https://warthogs.atlassian.net/browse/DPE-2573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -303,6 +303,10 @@ options:
     default: false
     type: boolean
     description: Enable timescaledb extension
+  plugin_audit_enable:
+    default: false
```
Comment (taurus-forever) [inline config.yaml:307]: JFYI, Audit plugin has been enabled on MySQL by default (for login actions only). Please check with @7annaba3l

**Review by taurus-forever — COMMENTED** [review]
Nice!

**Comment by taurus-forever** [comment]
Please update PR description: s/rock/snap/

**Review by lucasgameiroborges — APPROVED** [review]

**Comment by marceloneppel** [comment]
> Please update PR description: s/rock/snap/

Thanks, Alex! I updated the description.

**Review by dragomirp — APPROVED** [review]

**Comment by marceloneppel** [comment]
> Enable pgAudit by default

Thanks for pointing that out, Alex! After syncing with Mohamed, I updated the plugin to enable it by default on https://github.com/canonical/postgresql-operator/pull/612/commits/26ad2c09de88834b2533eec790bbdc4d4df6aa0b.

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -541,6 +541,9 @@ def test_enable_disable_extensions(harness, caplog):
   plugin_timescaledb_enable:
     default: false
     type: boolean
+  plugin_audit_enable:
+    default: false
```
Comment (taurus-forever) [inline tests/unit/test_charm.py:545]: We are testing possibility to enable pgaudit.
IMHO, we should keep default true, but test possibility of disabling it in the test below.

**Review by taurus-forever — APPROVED** [review]
LGTM, but I would invert the test logic to be close to the real scenarios.

File: tests/unit/test_charm.py (hunk 2 above)
Comment (marceloneppel) [reply tests/unit/test_charm.py]: Good catch! Fixed on https://github.com/canonical/postgresql-operator/pull/612/commits/e1d291686f486dc8bb5a3eea2b9dac1ad6f831c7.

**Pull request merged**